Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Linux.swift

```diff
@@ -0,0 +1,135 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Glibc)
+
+import Glibc
+import SystemPackage
+import FoundationEssentials
+package import _CShims
+
+// Linux specific implementations
+extension Subprocess.Configuration {
+    internal typealias StringOrRawBytes = Subprocess.StringOrRawBytes
+
+    internal func spawn(
+        withInput input: Subprocess.ExecutionInput,
+        output: Subprocess.ExecutionOutput,
+        error: Subprocess.ExecutionOutput
+    ) throws -> Subprocess {
+        let (executablePath,
+             env, argv,
+             intendedWorkingDir,
+             uidPtr, gidPtr,
+             supplementaryGroups
+        ) = try self.preSpawn()
+        defer {
+            for ptr in env { ptr?.deallocate() }
+            for ptr in argv { ptr?.deallocate() }
+            uidPtr?.deallocate()
+            gidPtr?.deallocate()
+        }
+
+        let fileDescriptors: [CInt] = [
+            input.getReadFileDescriptor().rawValue, input.getWriteFileDescriptor()?.rawValue ?? 0,
+            output.getWriteFileDescriptor().rawValue, output.getReadFileDescriptor()?.rawValue ?? 0,
+            error.getWriteFileDescriptor().rawValue, error.getReadFileDescriptor()?.rawValue ?? 0
+        ]
+
+        var workingDirectory: String?
+        if intendedWorkingDir != FilePath.currentWorkingDirectory {
+            // Only pass in working directory if it's different
+            workingDirectory = intendedWorkingDir.string
+        }
+        // Spawn
+        var pid: pid_t = 0
+        let spawnError: CInt = executablePath.withCString { exePath in
+            return workingDirectory.withOptionalCString { workingDir in
```

Comment:
I think `workingDirectory` and `executablePath` should be using `withFileSystemRepresentation` here instead of `withCString`